File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    var answer = -1;
    let stk = []
    for (let i = 0; i < s.length; i++) {
        if (stk.length > 0) {
            if (stk[stk.length-1] === s[i]){
                stk.pop()
            }
            else {
                stk.push(s[i])
            }   
        }
        else {
            stk.push(s[i])
        }
        
    }
    if (stk.length > 0) {
        answer = 0
    }
    else {
        answer = 1
    }

    return answer;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
class Solution
{
    public int solution(String s)
    {
        int answer = -1;
        StringBuilder sb = new StringBuilder();
        for (char k: s.toCharArray()) {
            if (sb.length() > 0 && sb.charAt(sb.length() - 1) == k) {
                sb.deleteCharAt(sb.length() - 1);
            } else {
                sb.append(k);
            }
        }
        
        if (sb.length() == 0) {
            answer = 1;
        } else {
            answer = 0;
        }

        return answer;
    }
}